src/lib/swapper/swappers/ThorchainSwapper/utils/longTailHelpers.ts

```diff
@@ -107,54 +112,62 @@ type ContractData = {
   tokenOut: Address
 }
 
-export const getContractDataByPool = async (
-  poolAddresses: Map<Address, FeeAmount>,
+export const getContractDataByPool = (
+  poolAddresses: Map<Address, { token0: Address; token1: Address; fee: FeeAmount }>,
   publicClient: PublicClient,
   tokenAAddress: string,
   tokenBAddress: string,
-): Promise<Map<Address, ContractData>> => {
+): Map<Address, ContractData> => {
   const poolContracts = new Map<Address, ContractData>()
-  await Promise.all(
-    Array.from(poolAddresses.entries()).map(async ([address, feeAmount]) => {
-      const poolContract = getContract({
-        abi: IUniswapV3PoolABI,
-        address,
-        publicClient,
-      })
-      try {
-        const [token0, token1]: [Address, Address] = await Promise.all([
-          poolContract.read.token0(),
-          poolContract.read.token1(),
-        ])
-
-        const tokenIn = token0 === tokenAAddress ? token0 : token1
-        const tokenOut = token1 === tokenBAddress ? token1 : token0
+  Array.from(poolAddresses.entries()).forEach(([address, { fee, token0, token1 }]) => {
```

Comment:
Ditto address terminology, can we use `_` or `current` prefix here, to be able to do so? 🙏🏽 

```suggestion
  Array.from(poolAddresses.entries()).forEach(([address, { fee, _token0Address, _token1Address }]) => {
```

Reply:
https://github.com/shapeshift/web/pull/5902/commits/4bde8e1a0bb1460511b5d4254c174975d04efeb4